omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/ConditionResource2_2.java

```diff
@@ -0,0 +1,238 @@
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getUpdatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("condition");
+		description.addRequiredProperty("patient");
+		description.addRequiredProperty("clinicalStatus");
+		description.addRequiredProperty("verificationStatus");
+		description.addRequiredProperty("onsetDate");
+		description.addRequiredProperty("voided");
+		
+		return description;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Condition getByUniqueId(String uuid) {
+		return conditionService.getConditionByUuid(uuid);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#delete(java.lang.Object,
+	 *      java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected void delete(Condition condition, String reason, RequestContext requestContext) throws ResponseException {
+		conditionService.voidCondition(condition, reason);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#newDelegate()
+	 */
+	@Override
+	public Condition newDelegate() {
+		return new Condition();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#save(java.lang.Object)
+	 */
+	@Override
+	public Condition save(Condition condition) {
+		return conditionService.saveCondition(condition);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#purge(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void purge(Condition condition, RequestContext requestContext) throws ResponseException {
+		conditionService.purgeCondition(condition);
+	}
+	
+	/**
+	 * @param condition - the condition to get the name of
+	 * @return condition's name
+	 */
+	@PropertyGetter("display")
+	public String getDisplayString(Condition condition) {
+		if (condition.getCondition() == null) {
+			return "";
+		} else {
+			if (condition.getCondition().getCoded() != null)
+				return condition.getCondition().getCoded().getName().getName();
+			
+			return condition.getCondition().getNonCoded();
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doGetAll(org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected PageableResult doGetAll(RequestContext context) {
+		String patientUuid = context.getRequest().getParameter("patient");
+		
+		Patient patient = Context.getPatientService().getPatientByUuid(patientUuid);
+		
+		List<Condition> activeConditions = new ArrayList<Condition>();
+		
+		if (patient != null)
```

Comment:
If patient is null, then you should not silently ignore this. Throw an exception